client/src/components/Search/SearchForm.js

```diff
@@ -9,61 +9,55 @@ const searchLabel = i18n._t('Admin.SEARCH', 'Search');
 const clearLabel = i18n._t('Admin.CLEAR', 'Clear');
 
 /**
- * Display an advanced filter form.
+ * Perform the callback when the enter key is pressed
+ * @param function callback
+ * @returns {function(*): void}
  */
-class SearchForm extends Component {
-  constructor(props) {
-    super(props);
-
-    this.handleKeyDown = this.handleKeyDown.bind(this);
-  }
-
-  /**
-   * Handle enter key submission in search form
-   *
-   * @param {KeyboardEvent} e
-   */
-  handleKeyDown(e) {
+function onEnter(callback) {
+  return (e) => {
     if (e.key === 'Enter') {
       e.preventDefault();
-      this.props.onSearch();
+      callback();
     }
-  }
-
-  render() {
-    const { visible, expanded, onSearch, onClear,
-      formSchemaUrl, id, identifier, clearable } = this.props;
+  };
+}
 
-    const loadForm = (visible || expanded);
+/**
+ * Display an advanced filter form.
+ */
+function SearchForm({
```

Comment:
Would prefer not have a refactor from class to function component at the same time as all the other stuff, it's a good thing to do just distracting.  No point changing it back though.

Reply:
From memory, I ended up refactoring this because I was trying to debug something in `<Search />`. In the end, it ended up being completely unrelated. Still the move to a functional component is an improvement here.